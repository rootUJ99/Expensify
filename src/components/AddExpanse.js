import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';
export class AddExpanse extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div className="container" >
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapDispathToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispathToProps)(AddExpanse);