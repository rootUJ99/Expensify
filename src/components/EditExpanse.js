import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
export class EditExpanse extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
        console.log('updated', expense);
    }
    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id })
        this.props.history.push('/');
    }
    render() {
        return (
            <div>
                <div className="page-header" >
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button
                    className="button2 button2--remove"
                    onClick={this.onRemove}
                > Remove Expense</button>
            </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};
const mapDispathToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});
export default connect(mapStateToProps, mapDispathToProps)(EditExpanse);