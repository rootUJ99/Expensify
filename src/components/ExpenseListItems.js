import React from 'react';
import {Link} from 'react-router-dom';
const ExpenseListItem = ({ id, dispatch, description, amount, createdAt})=>(
    <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <h4>{amount}-{createdAt}</h4>
       
    </div>
);

export default ExpenseListItem;
