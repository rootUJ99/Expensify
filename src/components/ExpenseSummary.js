import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    );
}
const mapStateToProps=(state)=>{
    const visiblityExpenses = selectExpenses(state.expenses,state.filters);
    return{
        expenseCount:visiblityExpenses.length,
        expensesTotal:selectExpensesTotal(visiblityExpenses)
    };
};
export default connect(mapStateToProps)(ExpenseSummary);
