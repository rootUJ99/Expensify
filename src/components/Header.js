import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';
const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="active-nav" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="active-nav">Create</NavLink>
        <NavLink to="/edit" activeClassName="active-nav">Edit</NavLink>
    </header>
);
export default Header;