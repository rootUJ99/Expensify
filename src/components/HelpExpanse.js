import React from 'react';
import ReactDOM from 'react-dom';
const HelpExpanse = () => (
    <div>
        this is help
    </div>
);
export default HelpExpanse;