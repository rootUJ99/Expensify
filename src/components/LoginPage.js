import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
        <h1>Expensify Application</h1>
            <button className="button" onClick={startLogin}>Login</button>
        </div>
    </div>
)
const matchDispatchProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});
export default connect(undefined, matchDispatchProps)(LoginPage);