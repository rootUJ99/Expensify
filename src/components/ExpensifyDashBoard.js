import React from 'react';
import ReactDOM from 'react-dom';
import ExpenseList from './ExpenseList';
import ExpenseListItems from './ExpenseListItems';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';
const ExpensifyDashBoard = () => (
    <div>
        <ExpenseSummary/>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);
export default ExpensifyDashBoard;