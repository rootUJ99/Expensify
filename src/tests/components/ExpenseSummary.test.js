import React from 'react'
import {shallow} from 'enzyme';
export default ExpenseSummary;
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should correctly render ExpenseSummary with 1 expense',()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={0} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpenseSummary with multiple expenses',()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={23} expensesTotal={2354155} />);
    expect(wrapper).toMatchSnapshot();
});