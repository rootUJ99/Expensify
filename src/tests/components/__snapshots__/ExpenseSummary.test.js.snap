// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $235.00
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2,354,155.00
  </h1>
</div>
`;
